static/js/message_fetch.js

```diff
@@ -54,7 +54,11 @@ function process_result(data, opts) {
         message_util.add_old_messages(messages, opts.msg_list);
     }
 
-    if (message_lists.current === message_list.narrowed && message_list.narrowed.empty()) {
+    if (
+        opts.msg_list === message_list.narrowed &&
+        message_lists.current === message_list.narrowed &&
+        message_list.narrowed.empty()
+    ) {
```

Comment:
Should instead be just `if (opts.msg_list === message_lists.current && message_lists.current.empty())`? 

I guess I'm not sure whether `narrow_banner.show_empty_narrow_message` is a useful thing to do if your "All messages" view is empty; one can likely reproduce that state to check by muting all of your streams.

Reply:
So currently, it's true that if one did go to the effort of removing all private messages and muting all existing topics/streams, then the "All messages" view will be blank without an empty narrow message.

I feel like if we want to have an empty narrow title for that view, then we should file a separate issue for that, because there are a bunch of live update issues that I came across while testing your suggestion above.

<details>
<summary>For example, it's easy to get into a state like this if one has multiple tabs open ...</summary>

![Screenshot from 2022-11-23 18-31-54](https://user-images.githubusercontent.com/63245456/203612024-599ce8d1-0e08-4dfb-99e8-6ea52c945292.png)
</details>

Then, again it's so unlikely that someone's "All messages" view would be completely empty, that maybe it would be fine?

I think the current change keeps the behavior we had prior to my changes in https://github.com/zulip/zulip/commit/16c45aaf427a48af430eb6af8c9335d48c12c1ae, and that's super important for existing links that go to narrowed views. Or if you click on a right-click on a link so that it opens in a new tab.

Reply:
> I feel like if we want to have an empty narrow title for that view, then we should file a separate issue for that, because there are a bunch of live update issues that I came across while testing your suggestion above.

Yeah, sounds good - I am pretty sure this is just not a case we've considered a goal in the past, in part because the onboarding process sends at least one private message, and probably few users delete those.